consensus/ibft/pos.go

```diff
@@ -204,13 +206,21 @@ func (pos *PoSMechanism) ShouldWriteTransactions(blockNumber uint64) bool {
 
 // getNextValidators is a helper function for fetching the validator set
 // from the Staking SC
-func (pos *PoSMechanism) getNextValidators(header *types.Header) (ValidatorSet, error) {
+func (pos *PoSMechanism) getNextValidators(header *types.Header) (validators.ValidatorSet, error) {
 	transition, err := pos.ibft.executor.BeginTxn(header.StateRoot, header, types.ZeroAddress)
 	if err != nil {
 		return nil, err
 	}
 
-	return staking.QueryValidators(transition, pos.ibft.validatorKeyAddr)
+	valAddrs, err := staking.QueryValidators(transition, pos.ibft.signer.Address())
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: PoS contract doesn't hold BLS PubKeys
```

Comment:
Is this a reference for a future feature ?

Reply:
Resolving because this issue is fixed in the next PR https://github.com/0xPolygon/polygon-edge/pull/649